### PR TITLE
[FEM] Implement a better dialog layout management for histogram

### PR DESCRIPTION
**Pull request by 0penBrain**
Relates to #6959 

An side advantage of using Tool flag is that dialog opens as a pure child of FC, without adding a tab in the OS task bar.

### CODE REVIEW

**Comment by donovaly** [comment]
Many thanks. This looks better and can go in.

Just 2 things:

Super cool would be the following:
- user opened the histogram
- user changed the result type in the FreeCAD panel and instead of that the histogram window is closed, it will be refreshed using the new result type. This way one can easily step through the different result types and compare e.g. Tresca with Mises stress distribution
I think I an capable of implementing this, but have no time right now. Since you are at it already, maybe you can do this?

Please keep commenting ;-). It often take me a while to figure out what code does when I try to fix things. I would even say more than 50% of the work to fix bugs is to find out what the existing code does. I commented in your code.

File: src/Mod/Fem/femtaskpanels/task_result_mechanical.py [hunk 1]
```diff
@@ -503,13 +503,15 @@ def result_selected(self, res_type, res_values, res_unit, res_title):
 
         if len(plt.get_fignums()) > 0:
             plt.close()
+        plt.ioff()
```
Comment (donovaly) [inline src/Mod/Fem/femtaskpanels/task_result_mechanical.py:506]: What does this do? Can you please add a comment?

File: src/Mod/Fem/femtaskpanels/task_result_mechanical.py [hunk 2]
```diff
@@ -503,13 +503,15 @@ def result_selected(self, res_type, res_values, res_unit, res_title):
...
         plt.hist(res_values, bins=50, alpha=0.5, facecolor="blue")
         plt.xlabel(res_unit)
         plt.title(translate("FEM","Histogram of {}").format(res_title))
         plt.ylabel(translate("FEM","Nodes"))
         plt.grid(True)
         fig_manager = plt.get_current_fig_manager()
-        fig_manager.window.setWindowFlags(QtCore.Qt.WindowStaysOnTopHint)    # stay ontop
+        fig_manager.window.setParent(FreeCADGui.getMainWindow())
+        fig_manager.window.setWindowFlag(QtCore.Qt.Tool)    # stay ontop
```
Comment (donovaly) [inline src/Mod/Fem/femtaskpanels/task_result_mechanical.py:514]: Can you please add a comment? For me as non-Pythionist it is not clear why setting a flag "QtCore.Qt.Tool" puts the window on top

Comment (0penBrain) [reply src/Mod/Fem/femtaskpanels/task_result_mechanical.py:514]: Nothing to do with Python. ;) It's pure Qt Window Flag : https://doc.qt.io/qt-5/qt.html#WindowType-enum
Will add a comment too.

**Comment by 0penBrain** [comment]
> Super cool would be the following:
> 
> * user opened the histogram
> * user changed the result type in the FreeCAD panel and instead of that the histogram window is closed, it will be refreshed using the new result type. This way one can easily step through the different result types and compare e.g. Tresca with Mises stress distribution
>   I think I an capable of implementing this, but have no time right now. Since you are at it already, maybe you can do this?

I thought the same, but I strongly believe it should be processed separately. ;) I has deeper impact on how things are implemented (architectured). I lack of time now too, but may give it a try lateR.

> Please keep commenting ;-). It often take me a while to figure out what code does when I try to fix things. I would even say more than 50% of the work to fix bugs is to find out what the existing code does. I commented in your code.

I would have said 75% :)

**Comment by 0penBrain** [comment]
@donovaly should be OK now

**Comment by donovaly** [comment]
Many thanks! Merging.

**Pull request merged**